webapp/channels/src/components/post_edit_history/post_edit_history.tsx

```diff
@@ -37,25 +42,67 @@ const renderThumbVertical = (props: Record<string, unknown>): JSX.Element => (
 const PostEditHistory = ({
     channelDisplayName,
     originalPost,
-    postEditHistory,
 }: PropsFromRedux) => {
+    const [postEditHistory, setPostEditHistory] = useState<Post[]>([]);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [isLoading, setIsLoading] = useState<boolean>(false);
     const scrollbars = useRef<Scrollbars | null>(null);
     const {formatMessage} = useIntl();
+    const retrieveErrorHeading = formatMessage({
+        id: 'post_info.edit.history.retrieveError',
+        defaultMessage: 'Unable to load edit history',
+    });
+    const retrieveErrorSubheading = formatMessage({
+        id: 'post_info.edit.history.retrieveErrorVerbose',
+        defaultMessage: 'There was an error loading the history for this message. Check your network connection or try again later.',
+    });
 
     useEffect(() => {
+        const fetchPostEditHistory = async () => {
+            try {
+                setIsLoading(true);
+                const history = await Client4.getPostEditHistory(originalPost.id);
+                setPostEditHistory(history);
+                setHasError(false);
+            } catch (error) {
+                setHasError(true);
+                setPostEditHistory([]);
+            } finally {
+                setIsLoading(false);
+            }
+        };
+        fetchPostEditHistory();
         scrollbars.current?.scrollToTop();
-    }, [originalPost, postEditHistory]);
+    }, [originalPost]);
+
+    useEffect(() => {
```

Comment:
Could these two `useEffect`'s be combined? I think we could have `fetchPostEditHistory` call `setPostEditHistory` and `setHasError` before it tries to fetch the history instead of having the two separate

Reply:
I had a feeling this would come up as a question/comment but I feel like the two useEffect hooks actually serve a purpose here, and in effect have replaced some now removed code in `editPost`.

First hook on `originalPost` 
- Handles the refresh/loading of edit history
- This hook is **not** on `id` to support reloading the history for an already open message
    - If the user edits the message they're currently viewing history for we want to reload the history **BUT** I don't think we should clear the loaded data and show the loading indicator again, it's more seamless to have it happen invisibly in this case.

Second hook on `originalPost.id`
- In the case we're actually asking to view a different post we want to unload the old history and errors
    - As mentioned above if we didn't change post id's we **don't** want to unload the old history as it will cause flashing behavior that is undesirable

So personally I think the two hooks serve different functional purposes, however if you still desire to have them combined into one that's possible so long as we're okay with sacrificing some user experience when it comes to "refreshing" the view on post edit.

Reply:
I see. That makes a ton of sense. I didn't think about handling those two cases differently, and I do agree that it'd be nicer to not show the loading indicator after the initial history has been loaded for a post.

I suppose there's a small chance that other changes to the post trigger the history to reload by depending on the whole `originalPost` object, but I can't imagine that'd happen often. If it does happen, it's likely to have little effect anyway